### PR TITLE
docs(contrib): Describe how to add a new package

### PR DESCRIPTION
**Pull request by epage**
@Muscraft is considering adding a crate and we felt it was best to not just answer the question of how to do so but to document it!

### CODE REVIEW

File: src/doc/contrib/src/implementation/packages.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+# New Packages
+
+## Steps
+
+Choose the relevant parent directory
+- `credential/` for credential-process related packages
+- `benches/` for benchmarking of cargo itself
+- `crates/` for everything esle
```
Comment (weihanglo) [inline src/doc/contrib/src/implementation/packages.md:8]:
```suggestion
- `crates/` for everything else
```

File: src/doc/contrib/src/implementation/packages.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+Run `cargo new <name>`
+- `<name>`:
+  - We tend to use `-` over `_`
+  - For internal APIs, to avoid collisions with third-party subcommands, we can use the `cargo-util-` prefix
+  - For xtasks, we use the `xtask-` prefix
+- `package.rust-version`
+  - Internal packages tend to have a policy of "latest" with a `# MSRV:1` comment to get it auto-updated
```
[the remark this answers is not in the document]
Comment (epage) [reply src/doc/contrib/src/implementation/packages.md]: I've gone ahead and wrote a whole MSRV policy section and linked to it.

File: src/doc/contrib/src/implementation/packages.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  - Ecosystem packages tend to have a policy of "N-2" with a `# MSRV:3` comment to get it auto-updated
+  - If the right choice is inherited from the workspace, feel free to keep it that way
+- If running without [cargo new automatically adding to workspace](https://github.com/rust-lang/cargo/pull/12779), add it as a workspace member if not already captured by a glob
+
+If its an xtask,
+- Add it to `.cargo/config.toml`s `[alias]` table
+- Mark `package.publish = false`
+
+If published, add the package to `publish.py`, in dependency order
```
[the remark this answers is not in the document]
Comment (epage) [reply src/doc/contrib/src/implementation/packages.md]: I was wanting to hold off on getting into that level of detail to be more resilient against change.  I have updated it.  Is the new version sufficient?

**Review by weihanglo — APPROVED** [review]
I am pretty okay with this doc update 👍🏾.
Thank you.

**Pull request merged**